[split from #3093] vmap SDPA tests parametrize CUDNN_ATTENTION (backend3) on XPU which has no cuDNN backend

### 🐛 Describe the bug

test_cases:
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_vmap_xpu.TestVmapBatchedGradientXPU,test_randomness_backend3_randomness_different_xpu
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_vmap_xpu.TestVmapBatchedGradientXPU,test_randomness_backend3_randomness_error_xpu
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_vmap_xpu.TestVmapBatchedGradientXPU,test_randomness_backend3_randomness_same_xpu
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_vmap_xpu.TestVmapBatchedGradientXPU,test_sdpa_backend3_xpu
---

## Parent Issue

Split from #3093 — XPU does not support NestedTensor for SDPA operations.

## Affected Test Cases

- `test_vmap_xpu.py::TestVmapBatchedGradientXPU::test_randomness_backend3_randomness_different_xpu`
- `test_vmap_xpu.py::TestVmapBatchedGradientXPU::test_randomness_backend3_randomness_error_xpu`
- `test_vmap_xpu.py::TestVmapBatchedGradientXPU::test_randomness_backend3_randomness_same_xpu`
- `test_vmap_xpu.py::TestVmapBatchedGradientXPU::test_sdpa_backend3_xpu`

## Root Cause

These four tests are parametrized over the `PLATFORM_SPECIFIC_SDPA` backend list at the top of `test/xpu/functorch/test_vmap_xpu.py`. Lines 109-110 unconditionally append `SDPBackend.CUDNN_ATTENTION` for XPU:

```python
# For XPU, add CUDNN_ATTENTION even though it's not supported - tests will fail with known issue
if TEST_XPU and SDPBackend.CUDNN_ATTENTION not in PLATFORM_SPECIFIC_SDPA:
    PLATFORM_SPECIFIC_SDPA.append(SDPBackend.CUDNN_ATTENTION)
```

This makes the `backend3` parametrization point at `SDPBackend.CUDNN_ATTENTION`. Each test wraps its SDPA call in `sdpa_kernel([backend])`, so all other backends (FLASH/EFFICIENT/MATH/OVERRIDEABLE) are disabled. XPU has no `_cudnn_attention_forward` registration (Python hook `torch._C._can_use_cudnn_attention` is `#ifdef USE_CUDA`-only in `torch/csrc/Module.cpp`), so the dispatcher rejects every backend and raises:

> RuntimeError: No viable backend for scaled_dot_product_attention was found. This is likely due to turning off both the math kernel and the overrideable kernels.

Inputs in these tests are **dense** tensors, so the failure happens in the C++ dense SDPA selector (not in `jagged_scaled_dot_product_attention`). The "overrideable" kernel hook is also not enabled for XPU SDPA in this code path.

This is a **test-side defect**: XPU should not advertise `CUDNN_ATTENTION` as a supported backend. The author of those lines was aware (`"# … tests will fail with known issue"`), but the test was not actually skipped for the cuDNN parametrization.

## Reproducer

```bash
cd third_party/torch-xpu-ops/test/xpu/functorch
PYTHONPATH=$(git rev-parse --show-toplevel)/test/functorch:$PYTHONPATH \
  python -m pytest -sxv test_vmap_xpu.py \
  -k "test_randomness_backend3_randomness_different_xpu or \
      test_randomness_backend3_randomness_error_xpu or \
      test_randomness_backend3_randomness_same_xpu or \
      test_sdpa_backend3_xpu"
```

(`PYTHONPATH` is needed so `from common_utils import …` at the top of `test_vmap_xpu.py` resolves to `pytorch/test/functorch/common_utils.py`.)

## Representative Error

```
test_vmap_xpu.py::TestVmapBatchedGradientXPU::test_randomness_backend3_randomness_different_xpu FAILED
=================================== FAILURES ===================================
_ TestVmapBatchedGradientXPU.test_randomness_backend3_randomness_different_xpu _
Traceback (most recent call last):
  File ".../test/xpu/functorch/test_vmap_xpu.py", line 3970, in test_randomness
    vmap(
  File ".../torch/_functorch/apis.py", line 220, in wrapped
    return vmap_impl(
  File ".../torch/_functorch/vmap.py", line 316, in vmap_impl
    return _flat_vmap(
  File ".../torch/_functorch/vmap.py", line 507, in _flat_vmap
    batched_outputs = func(*batched_inputs, **kwargs)
  File ".../test/xpu/functorch/test_vmap_xpu.py", line 3967, in f
    return F.scaled_dot_product_attention(q, k, v, dropout_p=dropout)
RuntimeError: No viable backend for scaled_dot_product_attention was found. This is likely due to turning off both the math kernel and the overrideable kernels.
```

## Recommended Fix Direction

- Either remove the unconditional `CUDNN_ATTENTION` append for XPU at `test_vmap_xpu.py:109-110`, or skip the `_backend3_*` parametrization for XPU (e.g. with `@skipXPUIf(True, "no cuDNN backend on XPU")` on `test_randomness` and `test_sdpa`, or by adding `_backend3_` to the `skip_list_common.py` entry — note the file is currently `None`-skipped there but the `op_ut` runner used to reproduce bypasses that).
- This is independent of the NestedTensor flash work in pytorch/pytorch#180857 and intel/torch-xpu-ops#3405; even after those land, XPU still has no cuDNN attention.

## Shared Context

- Reproduced 2026-04-30 on `pytorch` HEAD (current branch `main`) with `torch-xpu-ops` pinned at `d67a87783002c17786e9501c8c67b360479f6bb7` (file `third_party/xpu.txt`).
- Hardware: Intel(R) Data Center GPU Max 1550, 2 devices.
- Test runner: `pytest 7.3.2`, Python 3.12.3.

### Versions

PyTorch version: 2.13.0a0+git9b87139
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.4 LTS (x86_64)
GCC version: (Ubuntu 14.2.0-4ubuntu2~24.04.1) 14.2.0
Clang version: Could not collect
CMake version: version 3.31.6
Libc version: glibc-2.39

Python version: 3.12.3 (main, Mar  3 2026, 12:15:18) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.17.0-14-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: True
XPU used to build PyTorch: 20250303
Intel GPU driver version:
* libze1:       1.24.0.0-1146~24.04
* intel-opencl-icd:     25.18.33578.51-1146~24.04
Intel GPU models onboard:
* Intel(R) Data Center GPU Max 1550
Intel GPU models detected:
* [0] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD5, uuid=8680d50b-2f00-0000-8c00-000000000001, driver_version='1.6.33578+51', total_memory=65520MB, local_mem_size=128KB, max_compute_units=512, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=512, gpu_subslice_count=64, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [1] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD5, uuid=8680d50b-2f00-0000-8c00-000000000002, driver_version='1.6.33578+51', total_memory=65520MB, local_mem_size=128KB, max_compute_units=512, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=512, gpu_subslice_count=64, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           46 bits physical, 57 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  64
On-line CPU(s) list:                     0-63
Vendor ID:                               GenuineIntel
Model name:                              Intel(R) Xeon(R) Platinum 8352Y CPU @ 2.20GHz
CPU family:                              6
Model:                                   106
Thread(s) per core:                      2
Core(s) per socket:                      32
Socket(s):                               1
Stepping:                                6
CPU(s) scaling MHz:                      24%
CPU max MHz:                             3400.0000
CPU min MHz:                             800.0000
BogoMIPS:                                4400.00
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities
Virtualization:                          VT-x
L1d cache:                               1.5 MiB (32 instances)
L1i cache:                               1 MiB (32 instances)
L2 cache:                                40 MiB (32 instances)
L3 cache:                                48 MiB (1 instance)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-63
Vulnerability Gather data sampling:      Vulnerable
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Mitigation; Aligned branch/return thunks
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Old microcode:             Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Not affected

Versions of relevant libraries:
[pip3] flake8==6.1.0
[pip3] flake8-bugbear==23.3.23
[pip3] flake8-comprehensions==3.15.0
[pip3] flake8-executable==2.1.3
[pip3] flake8-logging-format==0.9.0
[pip3] flake8-pyi==23.3.1
[pip3] flake8-simplify==0.19.3
[pip3] intel-cmplr-lib-ur==2025.3.3
[pip3] intel-openmp==2025.3.3
[pip3] mkl==2024.2.0
[pip3] mkl-include==2024.2.0
[pip3] mkl-static==2024.2.0
[pip3] mypy==1.13.0
[pip3] mypy_extensions==1.1.0
[pip3] numpy==2.1.0
[pip3] onemkl-license==2025.3.1
[pip3] onnx==1.20.0
[pip3] onnx-ir==0.1.16
[pip3] onnxscript==0.6.2
[pip3] optree==0.13.0
[pip3] tbb==2021.13.1
[pip3] tbb-devel==2022.3.1
[pip3] tcmlib==1.4.1
[pip3] torch==2.13.0a0+git9b87139
[pip3] triton==3.7.0+git9c288bc5
[pip3] triton-xpu==3.7.1+git21033c4e
[pip3] umf==1.0.3

--- a/test/xpu/functorch/test_vmap_xpu.py
+++ b/test/xpu/functorch/test_vmap_xpu.py
@@ -105,10 +105,6 @@ def get_platform_specific_sdpa():
 
 PLATFORM_SPECIFIC_SDPA = get_platform_specific_sdpa()
 
-# For XPU, add CUDNN_ATTENTION even though it's not supported - tests will fail with known issue
-if TEST_XPU and SDPBackend.CUDNN_ATTENTION not in PLATFORM_SPECIFIC_SDPA:
-    PLATFORM_SPECIFIC_SDPA.append(SDPBackend.CUDNN_ATTENTION)
-
 FALLBACK_REGEX = "There is a performance drop"
 
 

